neurow/lib/neurow/broker/receiver_shard_manager.ex

```diff
@@ -35,20 +35,26 @@ defmodule Neurow.Broker.ReceiverShardManager do
     {:noreply, state}
   end
 
-  def all_pids(fun) do
+  def receiver_shards() do
     Enum.map(0..(@shards - 1), fn shard ->
-      fun.({shard, Neurow.Broker.ReceiverShard.build_name(shard)})
+      {shard, Neurow.Broker.ReceiverShard.build_name(shard)}
     end)
   end
 
   def rotate do
-    Stats.inc_history_rotate()
+    Neurow.Stats.MessageBroker.inc_history_rotate()
 
-    all_pids(fn {_, pid} ->
+    Enum.each(receiver_shards(), fn {_shard, pid} ->
       send(pid, {:rotate})
     end)
   end
 
+  def topic_count do
```

Comment:
**Suggestion** 🤔 : if we prefer the pipeline route, i would have write it like this

```
  receiver_shards()
  |> Enum.map(&Neurow.Broker.ReceiverShard.topic_count(&1.pid))
  |> Enum.sum()
```